### PR TITLE
Add conditions functionality to SubjectPermission object

### PR DESCRIPTION
**Pull request by sam-nguyen7**
Currently no conditions are being updated when ClusterRoleBindings and RoleBindings are being created. This PR adds the functionality to add/update conditions in the event of ClusterRoleBinding and/or RoleBinding creation.

### CODE REVIEW

**Comment by fahlmant** [comment]
Please squash

File: pkg/controller/subjectpermission/subjectpermission_controller.go [hunk 1]
```diff
@@ -118,13 +118,14 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
...
+		// build a clusterRoleBindingNameList which consists of clusterRoleName-subjectName
+		crClusterRoleBindingNameList := buildClusterRoleBindingCRList(instance)
 
-	// check ClusterRoleBindingName
-	populateCrClusterRoleBindingNameList := populateClusterRoleBindingNames(crClusterRoleBindingNameList, clusterRoleBindingList)
+		// check ClusterRoleBindingName
+		populateCrClusterRoleBindingNameList := populateClusterRoleBindingNames(crClusterRoleBindingNameList, clusterRoleBindingList)
 
-	if len(instance.Spec.ClusterPermissions) != 0 && len(populateCrClusterRoleBindingNameList) != 0 {
+		var counter int
```
Comment (jharrington22) [inline pkg/controller/subjectpermission/subjectpermission_controller.go]: `CreatedClusterRoleBindingCount` more descriptive?

File: pkg/controller/subjectpermission/subjectpermission_controller.go [hunk 2]
```diff
@@ -151,9 +152,22 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
...
+			err = r.client.Status().Update(context.TODO(), instance)
+			if err != nil {
+				reqLogger.Error(err, "Failed to update condition in subjectpermission controller when successfully created all cluster role bindings")
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{}, nil
 		}
 	}
-
+	reqLogger.Info("onto permission")
```
Comment (jharrington22) [inline pkg/controller/subjectpermission/subjectpermission_controller.go]: Debug line?

File: pkg/controller/subjectpermission/subjectpermission_controller.go [hunk 3]
```diff
@@ -162,39 +176,60 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	// compile list of allowed namespaces only for this subject permission. NOT a list of subject permissions
-	for _, permission := range instance.Spec.Permissions {
-
+	if len(instance.Spec.Permissions) != 0 {
+		var roleBindingCounter int
```
Comment (jharrington22) [inline pkg/controller/subjectpermission/subjectpermission_controller.go]: `CreatedRoleBindingCount` more descriptive?

File: pkg/controller/subjectpermission/subjectpermission_controller.go [hunk 4]
```diff
@@ -162,39 +176,60 @@ func (r *ReconcileSubjectPermission) Reconcile(request reconcile.Request) (recon
...
-			// get a list of all rolebindings in namespace
-			rbList := &v1.RoleBindingList{}
-			opts := client.ListOptions{Namespace: ns}
-			err = r.client.List(context.TODO(), &opts, rbList)
+		// compile list of allowed namespaces only for this subject permission. NOT a list of subject permissions
+		for _, permission := range instance.Spec.Permissions {
+			// list of all namespaces in safelist
+			safeList := controllerutil.GenerateSafeList(permission.NamespacesAllowedRegex, permission.NamespacesDeniedRegex, nsList)
+
+			var namespaceCounter int
```
Comment (jharrington22) [inline pkg/controller/subjectpermission/subjectpermission_controller.go]: `namespaceCount` to be consistent?

**Pull request merged**